carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/Screenshot.java

```diff
@@ -471,8 +483,7 @@ private static BufferedImage takeFullScreenshot(WebDriver driver, WebDriver augm
      * 
      * @return screenshot image
      */
-    private static BufferedImage takeVisibleScreenshot(WebDriver driver, WebDriver augmentedDriver) throws Exception {
-    	BufferedImage screenShot = ImageIO.read(((TakesScreenshot) augmentedDriver).getScreenshotAs(OutputType.FILE));
-        return screenShot;
+    private static BufferedImage takeVisibleScreenshot(WebDriver augmentedDriver) throws Exception {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS00112)